modules/airgridRtdProvider.js

```diff
@@ -62,54 +57,43 @@ export function getMatchedAudiencesFromStorage() {
   }
 }
 
-/**
- * Mutates the adUnits object
- * @param {Object} adUnits
- * @param {Array} audiences
- * @return {void}
- */
-function setAudiencesToAppNexusAdUnits(adUnits, audiences) {
-  adUnits.forEach((adUnit) => {
-    adUnit.bids.forEach((bid) => {
-      if (bid.bidder && bid.bidder === 'appnexus') {
-        deepSetValue(bid, 'params.keywords.perid', audiences || []);
-      }
-    });
-  });
-}
-
 /**
  * Pass audience data to configured bidders, using ORTB2
+ * @param {Object} bidConfig
  * @param {Object} rtdConfig
  * @param {Array} audiences
- * @return {{}} a map from bidder code to ORTB2 config
+ * @return {void}
  */
-export function setAudiencesAsBidderOrtb2(rtdConfig, audiences) {
+export function setAudiencesAsBidderOrtb2(bidConfig, rtdConfig, audiences) {
   const bidders = deepAccess(rtdConfig, 'params.bidders');
   if (!bidders || bidders.length === 0 || !audiences || audiences.length === 0) return;
 
-  const keywords = audiences.map(
+  const agOrtb2 = {};
+
+  const agKeywords = audiences.map(
     (audienceId) => `perid=${audienceId}`
   ).join(',');
+  deepSetValue(agOrtb2, 'user.keywords', agKeywords);
 
-  config.mergeBidderConfig({
-    bidders: bidders,
-    config: {
-      ortb2: {
-        site: {
-          keywords,
-        }
-      }
+  const agUserData = [
+    {
+      id: String(AG_TCF_ID),
+      name: 'airgrid',
+      segment: audiences.map((audienceId) => {
+        return {
+          id: audienceId,
+          name: `audience_${audienceId}`,
+          value: audienceId,
+        };
+      })
     }
-  })
-}
+  ]
+  deepSetValue(agOrtb2, 'user.data', agUserData);
 
-export function setAudiencesUsingAppNexusAuctionKeywords(audiences) {
-  config.setConfig({
-    appnexusAuctionKeywords: {
-      perid: audiences,
-    },
```

Comment:
@dgirardi Are you 100% certain that we should not be using the appnexus specific method - have they removed this in v8?

Reply:
Yes, however, this change would break the integration for v7 - so good catch. (I did not realize appnexus' keyword logic was only in the v8 branch). 

I'll rebase this PR against 8 soon - we are currently hoping to update the appnexus adapter to pick up `user.data` so I may try to account for that.

Reply:
it is still accessible to publishers but not rtd modules